### PR TITLE
Use constants for header names

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

**Comment by LucioFranco** [comment]
> making them associated constants for Status doesn't seem like the good place to do so, considering that Status itself is not a header

I actually don't agree, we convert these headers in Status and its good to encapsulate the encoding portion of these header names within the status.rs. So to me this makes sense. Metadata is much more abstract than specific extraction of headers. In addition, the metadata shouldn't contain the status headers etc that should all be extracted from the Status functions.

File: tonic/src/status.rs [hunk 1]
```diff
@@ -588,6 +584,13 @@ impl Status {
         self.add_header(response.headers_mut()).unwrap();
         response
     }
+
+    /// Header name for the `grpc-status` header
+    pub const GRPC_STATUS: HeaderName = HeaderName::from_static("grpc-status");
+    /// Header name for the `grpc-message` header
+    pub const GRPC_MESSAGE: HeaderName = HeaderName::from_static("grpc-message");
+    /// Header name for the `grpc-status-details-bin` header
+    pub const GRPC_STATUS_DETAILS: HeaderName = HeaderName::from_static("grpc-status-details-bin");
```
Comment (LucioFranco) [inline tonic/src/status.rs:593]: Im kinda torn on exposing these publically,I know we need to use them in tonic-web but I wonder if we could just do doc(hidden)?

Comment (djc) [reply tonic/src/status.rs:593]: `#[doc(hidden)]` seems fine to me, will change.

**Comment by tottoto** [comment]
@LucioFranco

>> making them associated constants for Status doesn't seem like the good place to do so, considering that Status itself is not a header
>
> I actually don't agree, we convert these headers in Status and its good to encapsulate the encoding portion of these header names within the status.rs. So to me this makes sense. Metadata is much more abstract than specific extraction of headers. In addition, the metadata shouldn't contain the status headers etc that should all be extracted from the Status functions.

I understand that the abstraction of metadata is a high level one. Then, I think it's fine to use Status's associated constants, considering that these will be `doc(hidden)` and will not actually be made into public API.

**Comment by djc** [comment]
Made them `#[doc(hidden)]` as requested.

**Review by LucioFranco — APPROVED** [review]

**Pull request merged**